### PR TITLE
Simple change I've been using in mosaic generation

### PR DESCRIPTION
**Pull request by ratbunch**
I've been lost in datasets for awhile but figured I'd share this change since I've seen some better variation when creating datasets. Fitted a curve to what you had previously which was a more "piecewise" function and added some slight randomness.

I've been mostly working with datasets and model improvements so forgive me if this is used elsewhere where it might mess something else up.

The later changes in constant you might not want. But I've seen more natural variability with these values along with my fit. Look at dataset generation and think "yep that's looks like JAV censoring" more often.

I tried different fits and ended with the sigmoid fit you can see below. Kept the linear increase you had previously above areas of 50k

4769e6cb-f813-48c1-927a-1fd5705eba91

Its a minor change but something I thought worth was sharing.

### CODE REVIEW

**Comment by ladaapp** [comment]
Hey, sorry for the late response.

That sounds promising, thanks for sharing! 

I will definitely give it a try and fine-tune the current model on my dataset :crossed_fingers:

If I can reproduce the improvement for small mosaics and hopefully don't notice regressions elsewhere it might be worth to train a mosaic size classification model. That probably shouldn't be too much effort, just minor adjustments to the mosaic detection model training and dataset creation script I suppose. Then we could plot your graph against real-world JAV.

It may will take a while until I come around retraining and testing this change properly so be prepared that this PR will probably stay open for a while. Sorry for that.

> I've been mostly working with datasets and model improvements so forgive me

Forgiven. :smile:

**Comment by ladaapp** [comment]
Hey, sorry for the delay. Started a run finetuning the current restoration model with your changes.
If it turns out well will do the same for the detection model :crossed_fingers:

**Comment by ladaapp** [comment]
Finished training of the detection and the restoration model on a small test dataset. For the restoration model I didn't see a noticeable change. The mosaic detection model showed a subtle improvement on certain patterns I think.

As it didn't show a regression and potential improvement, let's merge it and make it the new default :)

I see that you changed the v1 method `get_mosaic_block_size_v1`. Could you please create a new versioned method instead and adjust `restoration_dataset_metadata.py`?

**Comment by ladaapp** [comment]
Closing this MR as we've moved the project to [Codeberg](https://codeberg.org/ladaapp/lada). Feel free to reopen your PR there! Thank you for your contribution.
